scripts/earthengine/earthengine_image.py

```diff
@@ -649,29 +634,54 @@ def ee_process(config) -> list:
             if config.get('ee_export_image', True):
                 task = export_ee_image_to_gcs(img, config)
                 ee_tasks.append(task)
+                counters.add_counter('total', 1)
         else:
             logging.error(f'Failed to generate image for config: {config}')
         # Advance time to next period.
         time_period = config.get('time_period', 'P1M')
         for ts in ['start_date', 'end_date']:
-            dt = _advance_date(config.get(ts, ''), time_period)
+            dt = utils.date_advance_by_period(config.get(ts, ''), time_period)
             if dt:
                 config[ts] = dt
         config['ee_image_count'] = config['ee_image_count'] - 1
         logging.info(f'Advanced dates by {time_period}: {config}')
 
+    completed_tasks = []
+    if not config.get('ee_wait_task', True):
+        return ee_tasks
+
     # Wait for tasks to complete
-    if config.get('ee_wait_task', True):
-        pending_tasks = set(ee_tasks)
-        while len(pending_tasks) > 0:
-            task = pending_tasks.pop()
-            if task.active():
-                pending_tasks.add(task)
-                logging.info(f'Waiting for task: {task}')
-                time.sleep(10)
-            else:
-                task_status = ee.data.getTaskStatus(task.id)
-                logging.info(f'EE task completed: {task_status}')
+    pending_tasks = set(ee_tasks)
+    while len(pending_tasks) > 0:
+        task = pending_tasks.pop()
+        if task.active():
+            pending_tasks.add(task)
+            logging.info(f'Waiting for task: {task}')
+            time.sleep(10)
+        else:
+            task_status = ee.data.getTaskStatus(task.id)
+            logging.info(f'EE task completed: {task_status}')
+            for status in task_status:
+                # Get the destination file for each completed task.
+                image_file = status.get('description', '')
+                gcs_path = status.get('destination_uris', [])[0]
+                if gcs_path and image_file:
+                    gcs_path = re.sub('https.*storage/browser/', 'gs://',
+                                      gcs_path)
+                    status['output_file'] = f'{gcs_path}{image_file}.tif'
+                completed_tasks.append(status)
+            counters.add_counter('processed', 1)
+    return completed_tasks
+
+
+def get_date_from_filename(filename: str, time_period: str = '') -> str:
+    '''Returns the date from the filename.'''
+    matches = re.search(r'from_(?P<date>[0-9]{4}[0-9-]{2}[0-9-]{0,2})',
+                        filename)
+    if matches:
+        return utils.date_format_by_time_period(
+            matches.groupdict().get('date', ''), time_period)
+    return ''
```

Comment:
nit: this feels like a lot of magic :) so a couple of comments would help.

Reply:
Added some comments.